openapi-core/src/main/java/io/ballerina/openapi/core/generators/client/ClientMetaData.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2022, WSO2 LLC. (http://www.wso2.com). All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.ballerina.openapi.core.generators.client;
+
+import io.ballerina.openapi.core.model.Filter;
+import io.swagger.v3.oas.models.OpenAPI;
+
+/**
+ * This class stores metadata that related to client code generations.
+ *
+ * @since 1.3.0
+ */
+public class ClientMetaData {
```

Comment:
Any reason to duplicate this class in both code generator and IDL modules?

Reply:
Yeah, This contains client generation metadata and IDL one used to extract annotation detail and store it. will try to move to one model.

Reply:
Fixed with this commit https://github.com/ballerina-platform/openapi-tools/pull/1071/commits/7f0134da0c2029c6bac2e469b9f16df098224521